monitor.cpp

```diff
@@ -3,36 +3,57 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <bitset>
+
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
+void DisplayCriticalIdicator() {
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+    cout << std::endl;
+}
+
+bool IsTemperatureInFarenheitCritical(float temperature) {
+    if (temperature < Min_Normal_Temperature_F || temperature > Max_Normal_Temperature_F) {
+        cout << "Temperature is critical!\n";
+        return true;
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+    return false;
+}
+
+bool IsPulseRateCritical(float pulseRate) {
+    if (pulseRate < Min_Normal_PulseRange || pulseRate > Max_Normal_PulseRange) {
```

Comment:
The range comparison repeats from the temperature (though the tool doesn't point it out). Changing the range logic (e.g., from < to <=) will need modification in multiple places.
Try extracting the range comparison logic

Reply:
Updated the logic